src/lib_db.ml

```diff
@@ -144,33 +144,36 @@ let create findlib ~scopes internal_libraries =
 let internal_libs_without_non_installable_optional_ones t =
   String_map.values t.instalable_internal_libs
 
+let interpret_lib_dep t ~dir lib_dep =
+  match lib_dep with
+  | Lib_dep.Direct name -> begin
+      match find_exn t ~from:dir name with
+      | x -> Inl [x]
+      | exception e ->
+        (* Call [find] again to get a proper backtrace *)
+        Inr { fail = fun () -> ignore (find_exn t ~from:dir name : Lib.t); raise e }
+    end
+  | Select { choices; loc; _ } ->
+    match
+      List.find_map choices ~f:(fun { required; forbidden; _ } ->
+        if String_set.exists forbidden ~f:(lib_is_available t ~from:dir) then
+          None
+        else
+          match
+            List.map (String_set.elements required) ~f:(find_exn t ~from:dir)
+          with
+          | l           -> Some l
+          | exception _ -> None)
```

Comment:
Unrelated: Maybe we should just add `Option.try_with` I see that we do this idiom in a few places already.